src/core.c/CompUnit/Repository/Installation.pm6

```diff
@@ -103,9 +110,12 @@ sub MAIN(:$name, :$auth, :$ver, *@, *%) {
 
             %hash;
         }
-        method content($content-id --> IO::Handle:D) { self!dist.content($content-id) }
-        method Str { CompUnit::Repository::Distribution.new(self).Str }
-        method id { $.dist-id }
+        method content(::?CLASS:D: $content-id --> IO::Handle:D) { self!dist.content($content-id) }
+        method module-dependency(::?CLASS:D: Str:D $module_name) is implementation-detail {
```

Comment:
I wonder why this isn't implemented on `CompUnit::Repository::Distribution` and only called on such objects, because this function should always have the same implementation regardless of the distribution or repository backing it.

Reply:
I don't remember the exact details, but somewhere `LazyDistribution` was used directly. But as you can see, all the method does – it re-delegates to CURD. `content` re-delegates to the backing distribution because that's what CURD does.

I can try removing them and recover what was wrong about them being missing. But I can guarantee that they are there not because I just wanted it be that way.

Reply:
@ugexe I wonder if `candidates` method is supposed to always return CURD. Because the reason for `module-dependency` method is `LazyDistribution` is returned by `CUR::Installation.candidates`.

Reply:
Addressed with b92ba9af8ff903da5c6da4a04ab3d40c49d219ef. I decided that it would be right to make all `candidates` method consistently return CURD.

Reply:
The idea at least is that the user shouldn't really care what implementation of Distribution they get, only that they get a `Distribution`. Its entirely possible people (maybe even myself) are abusing that by assuming an e.g. `Distribution::Hash` vs `Distribution::Path`, but it would probably be better to just fix any such instances.